server/src/services/InitGoogle.ts

```diff
@@ -159,14 +163,35 @@ async function createCredentialedClient() {
 // It's not necessary to recreate the client for each request, but it is safer.
 // If multiple tokens end up being used, it will be important to use the
 // appropriate one for that request.
-export async function createCalendarApi() {
-  const auth = await createCredentialedClient();
-  return calendar({ version: 'v3', auth });
+export async function createCalendarApi(
+  createClient: () => Promise<OAuth2Client> = async () =>
+    createCredentialedClient(
+      await prisma.google_tokens.findFirstOrThrow({
+        where: { is_valid: true },
+      }),
+    ),
+) {
+  return calendar({ version: 'v3', auth: await createClient() });
```

Comment:
```suggestion
export async function createCalendarApi(tokenInfo?: TokenInfo) {
  const tokens =
    tokenInfo ??
    (await prisma.google_tokens.findFirstOrThrow({
      where: { is_valid: true },
    }));
  const auth = await createCredentialedClient(tokens);
  return calendar({ version: 'v3', auth });
```
I found passing around `createClient` a little confusing and this approach seems more direct to me.